core/engine/src/builtins/temporal/plain_year_month/mod.rs

```diff
@@ -202,55 +208,153 @@ impl BuiltInConstructor for PlainYearMonth {
     }
 }
 
-// ==== `PlainYearMonth` Accessor Implementations ====
+// ==== `Temporal.PlainYearMonth` static Methods ====
 
 impl PlainYearMonth {
-    fn get_calendar_id(_this: &JsValue, _: &[JsValue], _: &mut Context) -> JsResult<JsValue> {
-        Err(JsNativeError::error()
-            .with_message("not yet implemented.")
-            .into())
+    // 9.2.2 `Temporal.PlainYearMonth.from ( item [ , options ] )`
+    fn from(_: &JsValue, args: &[JsValue], context: &mut Context) -> JsResult<JsValue> {
+        let item = args.get_or_undefined(0);
+        // 1. If Type(item) is Object or Type(item) is String and item is not null, then
+
+        let inner = if item.is_object() {
+            // 9.2.2.2
+            if let Some(data) = item
+                .as_object()
+                .and_then(JsObject::downcast_ref::<PlainYearMonth>)
+            {
+                // Perform ? [GetTemporalOverflowOption](https://tc39.es/proposal-temporal/#sec-temporal-gettemporaloverflowoption)(options).
+                let options = get_options_object(args.get_or_undefined(1))?;
+                let _ = get_option::<ArithmeticOverflow>(&options, js_str!("overflow"), context)?;
+                data.inner.clone()
+            } else {
+                let options = get_options_object(args.get_or_undefined(1))?;
+                let overflow = get_option(&options, js_str!("overflow"), context)?
+                    .unwrap_or(ArithmeticOverflow::Constrain);
+
+                // a. Let calendar be ? ToTemporalCalendar(item).
+                let calendar = to_temporal_calendar_slot_value(args.get_or_undefined(1))?;
+                InnerYearMonth::new(
+                    item.get_v(js_str!("year"), context)
+                        .expect("Year not found")
+                        .to_i32(context)
+                        .expect("Cannot convert year to i32"),
+                    item.get_v(js_str!("month"), context)
+                        .expect("Month not found")
+                        .to_i32(context)
+                        .expect("Cannot convert month to i32"),
+                    item.get_v(js_str!("day"), context)
+                        .map_or(Some(1), |x| x.to_i32(context).ok()),
+                    calendar,
+                    overflow,
+                )?
+            }
+        } else if item.is_string() {
```

Comment:
```suggestion
        } else if let Some(item_as_string) = item.as_string() {
```

Is something like this possible that way an expect could be avoided.